src/util.rs

```diff
@@ -37,57 +40,43 @@ pub fn vector_with_length<F: FieldElement>(len: usize) -> Vec<F> {
 }
 
 /// Unpacked proof with subcomponents
-#[derive(Debug)]
-pub struct UnpackedProof<'a, F: FieldElement> {
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(bound(deserialize = ""))]
+pub struct UnpackedProof<F: FieldElement> {
     /// Data
-    pub data: &'a [F],
+    pub data: Vec<F>,
     /// Zeroth coefficient of polynomial f
-    pub f0: &'a F,
+    pub f0: F,
     /// Zeroth coefficient of polynomial g
-    pub g0: &'a F,
+    pub g0: F,
     /// Zeroth coefficient of polynomial h
-    pub h0: &'a F,
+    pub h0: F,
     /// Non-zero points of polynomial h
-    pub points_h_packed: &'a [F],
+    pub points_h_packed: Vec<F>,
 }
 
 /// Unpacked proof with mutable subcomponents
-#[derive(Debug)]
-pub struct UnpackedProofMut<'a, F: FieldElement> {
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(bound(deserialize = ""))]
+pub struct UnpackedProofMut<F: FieldElement> {
     /// Data
-    pub data: &'a mut [F],
+    pub data: Vec<F>,
```

Comment:
I'm not sure if this is the right way to go, but trying to set the correct deserialization trait here with the struct lifetime and the deserialization lifetime was a bit much for me. I had to annotate each of these fields as so: `[serde(bound(deserialize="&'a mut F: Deserialize<'a>"))]`.  I thought it would be would be nice in general to replace serialization helpers with bincode wherever possible.